### PR TITLE
feat(backend): add quiz questions to db bb-59

### PR DESCRIPTION
**Pull request by luiqor**
### Add database migrations:
- quiz_questions table (with a label and a reference to categories table)
- quiz_questions data
- quiz_answers table (with a label, value and a reference to questions table)
- quiz_answers data

### CODE REVIEW

File: apps/backend/src/db/migrations/20240821161830_add_quiz_questions_data.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	[ColumnName.CATEGORY_ID]: number;
+	[ColumnName.LABEL]: string;
+};
+
+const TableName = {
+	CATEGORIES: "categories",
+	QUIZ_QUESTIONS: "quiz_questions",
+} as const;
+
+const CATEGORIES = {
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240821161830_add_quiz_questions_data.ts:18]: B23

File: apps/backend/src/db/migrations/20240821161830_add_quiz_questions_data.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+const ColumnName = {
+	CATEGORY_ID: "category_id",
+	CREATED_AT: "created_at",
+	ID: "id",
+	LABEL: "label",
+	NAME: "name",
+	UPDATED_AT: "updated_at",
+} as const;
+
+const QUIZ_QUESTIONS = {
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240821161830_add_quiz_questions_data.ts:38]: This is a map - from category to questions

File: apps/backend/src/db/migrations/20240822055434_add_quiz_answers_data.ts [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+	ID: "id",
+	LABEL: "label",
+	QUESTION_ID: "question_id",
+	VALUE: "value",
+} as const;
+
+const VALUE_START_INDEX_OFFSET = 1;
+const VALUE_MULTIPLIER = 2;
+
+const QUIZ_ANSWERS = {
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240822055434_add_quiz_answers_data.ts:29]: This is a map

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/backend/src/db/migrations/20240821161830_add_quiz_questions_data.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+	[ColumnName.CATEGORY_ID]: number;
+	[ColumnName.LABEL]: string;
+};
+
+const TableName = {
+	CATEGORIES: "categories",
+	QUIZ_QUESTIONS: "quiz_questions",
+} as const;
+
+const Categories = {
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240821161830_add_quiz_questions_data.ts:18]:
```suggestion
const Category = {
```

File: apps/backend/src/db/migrations/20240822004833_add_quiz_answers_table.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+} as const;
+
+function up(knex: Knex): Promise<void> {
+	return knex.schema.createTable(TableName.QUIZ_ANSWERS, (table) => {
+		table.increments(ColumnName.ID).primary();
+		table.text(ColumnName.LABEL).notNullable();
+		table
+			.integer(ColumnName.QUESTION_ID)
+			.references(ColumnName.ID)
+			.inTable(TableName.QUIZ_QUESTIONS)
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240822004833_add_quiz_answers_table.ts:29]: Let's add CASCADE delete strategy here

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/backend/src/db/migrations/20240821161830_add_quiz_questions_data.ts (hunk 4 above)
Comment (luiqor) [reply apps/backend/src/db/migrations/20240821161830_add_quiz_questions_data.ts]: 'Category' is already defined as type Category. I've changed `const Category = {` to more specific `const CategoryName = {`

File: apps/backend/src/db/migrations/20240822004833_add_quiz_answers_table.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+	UPDATED_AT: "updated_at",
+	VALUE: "value",
+} as const;
+
+const ValueLimits = {
+	MAX: 10,
+	MIN: 1,
+} as const;
+
+const CASCADE = "CASCADE";
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240822004833_add_quiz_answers_table.ts:22]:
```suggestion
const DELETE_STRATEGY = "CASCADE";
```

File: apps/backend/src/db/migrations/20240821161628_add_quiz_questions_table.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+function up(knex: Knex): Promise<void> {
+	return knex.schema.createTable(TableName.QUIZ_QUESTIONS, (table) => {
+		table.increments(ColumnName.ID).primary();
+		table.text(ColumnName.LABEL).notNullable().unique();
+		table
+			.integer(ColumnName.CATEGORY_ID)
+			.references(ColumnName.ID)
+			.inTable(TableName.CATEGORIES)
+			.notNullable();
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240821161628_add_quiz_questions_table.ts:24]: Add delete strategy

File: apps/backend/src/db/migrations/20240822004833_add_quiz_answers_table.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+const ColumnName = {
+	CREATED_AT: "created_at",
+	ID: "id",
+	LABEL: "label",
+	QUESTION_ID: "question_id",
+	UPDATED_AT: "updated_at",
+	VALUE: "value",
+} as const;
+
+const ValueLimits = {
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240822004833_add_quiz_answers_table.ts:17]:
```suggestion
const ValueLimit = {
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by revenets — APPROVED** [review]

**Review by fshabanov — APPROVED** [review]

**Comment by fshabanov** [comment]
Merge the main branch into yours and we can merge

**Pull request merged**